source/backend/dal/Repositories/AcquisitionFileRepository.cs

```diff
@@ -176,6 +176,12 @@ public short GetRegion(long id)
                 .FirstOrDefault() ?? throw new KeyNotFoundException();
         }
 
+        public PimsAcquisitionFile GetByProductId(long productId)
```

Comment:
This would fall into the TryGet pattern I believe since it returns null when a result is not found.
Also, what happens if you pass a null object to the mapper? it looks like this gets passed all the way back to the controller with no null checking, not sure if that would result in a null pointer or not.

Reply:
it does check for nulls. If null, undefined or empty it assumes there are no files associated with the product